packages/scoutgame/src/points/getPartnerRewards.ts

```diff
@@ -0,0 +1,121 @@
+import { prisma } from '@charmverse/core/prisma-client';
+import type { Season } from '@packages/dates/config';
+import { getCurrentSeasonStart, getSeasonWeekFromISOWeek } from '@packages/dates/utils';
+
+type PartnerRewardBase = {
+  week: number;
+  points: number;
+  type: 'optimism_new_scout_partner' | 'optimism_top_referrer';
+  season: Season;
+};
+
+export type OptimismNewScoutPartnerReward = PartnerRewardBase & {
+  type: 'optimism_new_scout_partner';
+  partner: string;
+  position: number;
+};
+
+export type OptimismTopReferrerReward = PartnerRewardBase & {
+  type: 'optimism_top_referrer';
+  partner: string;
+  date: string;
+};
+
+export type PartnerReward = OptimismNewScoutPartnerReward | OptimismTopReferrerReward;
+
+export async function getUnclaimedPartnerRewards({ userId }: { userId: string }) {
+  const partnerRewards = await prisma.partnerRewardPayout.findMany({
+    where: {
+      payoutContract: {
+        season: getCurrentSeasonStart()
+      },
+      userId,
+      claimedAt: null
+    },
+    select: {
+      id: true,
+      amount: true,
+      payoutContract: {
+        select: {
+          partner: true
+        }
+      }
+    }
+  });
+
+  return partnerRewards.map((payout) => ({
+    id: payout.id,
+    amount: payout.amount,
+    partner: payout.payoutContract.partner
+  }));
+}
+
+export async function getPartnerRewards({
+  userId,
+  isClaimed,
+  season
+}: {
+  userId: string;
+  isClaimed: boolean;
+  season: string;
+}) {
+  const optimismNewScoutPartnerRewards: OptimismNewScoutPartnerReward[] = [];
```

Comment:
suggestion: you just need one array. in this function's return response you combine them together and mix up the results by sorting by points anyway